proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,136 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`. The
+resource type is not a `RawBuffer` variant, and so there is a distinct set
+of rules that define valid RETs for this resource type.
+
+RETs for non-`RawBuffer` variants may include basic types (ints and uints of sizes 16
+and 32, as well as half, float, vectors, and matrices of 4 elements or fewer).
+Structs that contain fields of these basic types (where all fields in the struct have
+the same type) may also be RETs. 
+Structs that either have structs as fields or arrays of structs as fields may also be
+allowed, as long as there are at most 4 sub elements, and each sub element is at most
+32 bits. Additionally, resource types are not allowed within an RET, even if the 
+underlying resource type has a primitive RET (i.e., `RWBuffer<int>` as an RET).

```

Comment:
> even if the underlying resource type has a primitive RET (i.e., `RWBuffer<int>` as an RET).

What is a "primitive RET"?  I think all this sentence has to say is "Resource types are not allowed."  

If it really is important for some reason to say something about the RET of a resource that someone is attempting to use as a RET then maybe "Resource types are not allowed, regardless of that resource's RET."  But I don't think it is.

Reply:
Simplified.